doc/source_list_library.tex

```diff
@@ -1,40 +1,42 @@
 \section*{Appendix: List library}
 
-Those list library functions that are not builtins are pre-declared as follows:
+Those list library functions that are not built-ins are pre-declared as follows:
 
 \begin{lstlisting}
-// is_list recurses down the list and checks that it ends with the empty list []
+// is_list recurses down the list and checks that it ends with the empty list null
 
 function is_list(xs) {
-    return is_empty_list(xs) || (is_pair(xs) && is_list(tail(xs)));
+    return is_null(xs) || (is_pair(xs) && is_list(tail(xs)));
 }
 
 // equal computes the structural equality 
 // over its arguments
 
-function equal(item1, item2){
-    return (is_pair(item1) && is_pair(item2))
-        || (is_empty_list(item1) && is_empty_list(item2))
-        || item1 === item2;
+function equal(x, y){
+    return (is_pair(x) && is_pair(y)) 
+        ? (equal(head(x), head(y)) &&
+           equal(tail(x), tail(y)))
+        : (is_null(x) && is_null(y))
+          || x === y;
 }
 
 // returns the length of a given argument list
 // assumes that the argument is a list
 
 function length(xs) {
-    return is_empty_list(xs) 
+    return is_null(xs) 
 	? 0
-        : length(tail(xs));
+        : 1 + length(tail(xs));
 }
 
 // map applies first arg f, assumed to be a unary function,
 // to the elements of the second argument, assumed to be a list.
 // f is applied element-by-element: 
-// map(f, [1, [2, []]]) results in [f(1), [f(2), []]]
+// map(f, [1, [2, null]]) results in [f(1), [f(2), null]]
```

Comment:
Should replace `[1, [2, null]]` with `list(1, 2)` as per `append` definition for consistency? I recommend the `append` way as the current example can't be run in the playground in Source 1 and 2 as they don't support array syntax.  And maybe replace the result with `list(f(1), f(2)` also.

Reply:
noted!